### DISCUSSION TITLE
MAPIE's `sample_weight` vs sklearn's `sample_weight` in the `fit` method

### DISCUSSION THREAD

**Discussion by CarlaFernandez** [scikit-learn-contrib/MAPIE#362] [Q&A] [answered]
How do these parameters relate?
If my original model relied on sample_weight, can I simply wrap MAPIE around it and it would be the same internally?
Like this:
```python
self.mapie_regressor = MapieRegressor(estimator=self.model, method='plus', cv=5, random_state=self.seed)
        
self.mapie_regressor.fit(
    X=self.X_train, 
    y=self.y_train, 
    sample_weight=self.weights
)
```

**Comment by thibaultcordier** [comment, marked as answer]
Hello @CarlaFernandez,

TL;DR: These parameters are the same for MAPIE and scikit-learn in `fit` method.

Indeed, `MapieRegressor` or other MAPIE classes have been implemented to be scikit-learn compatible with commonly used terms. In particular, `sample_weight` is used in MAPIE specifically in the utility function [`fit_estimator`](https://github.com/scikit-learn-contrib/MAPIE/blob/master/mapie/utils.py#L79C3-L79C3) that depends on your model estimator.

In other words, if you simply wrap MAPIE around your original model which relies on `sample_weight`, it will be the same internally (direct transmission of `sample_weight` parameter).

Some additionnal links:
- https://scikit-learn.org/stable/glossary.html#term-sample_weight
- https://scikit-learn.org/stable/modules/generated/sklearn.base.RegressorMixin.html
- https://scikit-learn.org/stable/modules/generated/sklearn.base.ClassifierMixin.html
- https://stackoverflow.com/questions/54093185/how-to-understand-sample-weight-in-sklearn-metrics
- https://github.com/scikit-learn-contrib/MAPIE/blob/master/mapie/utils.py#L79C3-L79C3

**Comment by CarlaFernandez** [comment]
Thanks for the confirmation!
